### ISSUE TITLE
Why do subsequent sockets cause 'Segmentation fault: 11' when trying to disconnect?

### ISSUE THREAD

**Issue by colingourlay** [colingourlay/tcp-ping-sync#1]
Now that we're optionally using [netlinkwrapper][netlinkwrapper] sockets if the environment can build it, there's a segfault occuring when running the tests.

The first socket created is always successful and can make as many connections as it wants, but as soon as you create a second socket, a segfault will consistently occur on either the first or second connection it tries to make. It might have something to do with how rapidly those sockets are created, or maybe I'm not destroying them correctly, but I'm really not sure.

I don't know enough about how [netlinkwrapper][netlinkwrapper] (which wraps a very old release of [netLink][netlink]) works, so I'm kind of adrift here. I think the issue might have been fixed in [netLink][netlink], but hasn't been pulled into [netlinkwrapper][netlinkwrapper] (as the codebases look very different now).

I know this would be extremely low on your priority list, @JacobFischer, but if you have any thoughts, i'd appreciate you sharing them!

I've released a [beta](https://www.npmjs.com/package/tcp-ping-sync) to npm of the [`optional-netlinkwrapper`](https://github.com/colingourlay/tcp-ping-sync/tree/optional-netlinkwrapper) branch which includes this bug, as it's fit for use in projects that only create one socket, but I'd like to fix this before putting out the next major bump.

[netlinkwrapper]: https://github.com/JacobFischer/netlinkwrapper
[netlink]: https://github.com/Lichtso/netLink

**Comment by colingourlay** [comment]
I've added [segfault-handler]() to the tests, to try to get better output. Here's what it produced:

```sh
$ npm test -- --debug

> tcp-ping-sync@2.0.0-beta.1 test /code/tcp-ping-sync
> node test.js "--debug"

localPingA { results:
   [ { seq: 0, time: 2.157764 },
     { seq: 1, time: 0.490953 },
     { seq: 2, time: 0.413754 },
     { seq: 3, time: 0.451327 },
     { seq: 4, time: 0.368862 },
     { seq: 5, time: 0.507221 },
     { seq: 6, time: 0.408683 },
     { seq: 7, time: 0.388707 },
     { seq: 8, time: 0.44326 },
     { seq: 9, time: 0.395122 } ],
  stats:
   { min: 0.368862,
     avg: 0.6025653,
     max: 2.157764,
     stddev: 0.5201026264513667 },
  settings: { host: 'localhost', port: 2697, attempts: 10 } }
PID 14256 received SIGSEGV for address: 0x4c
0   segfault-handler.node               0x00000001021eb3e8 _ZL16segfault_handleriP9__siginfoPv + 312
1   libsystem_platform.dylib            0x00007fff7c14db3d _sigtramp + 29
2   ???                                 0x0000000000000000 0x0 + 0
3   netlinksocket.node                  0x00000001050a85d7 _ZN14NetLinkWrapper10DisconnectERKN2v820FunctionCallbackInfoINS0_5ValueEEE + 91
4   node                                0x000000010018a5c2 _ZN2v88internal25FunctionCallbackArguments4CallEPFvRKNS_20FunctionCallbackInfoINS_5ValueEEEE + 466
5   node                                0x00000001001ead20 _ZN2v88internal12_GLOBAL__N_119HandleApiCallHelperILb0EEENS0_11MaybeHandleINS0_6ObjectEEEPNS0_7IsolateENS0_6HandleINS0_10HeapObjectEEESA_NS8_INS0_20FunctionTemplateInfoEEENS8_IS4_EENS0_16BuiltinArgumentsE + 816
6   node                                0x00000001001ea360 _ZN2v88internalL26Builtin_Impl_HandleApiCallENS0_16BuiltinArgumentsEPNS0_7IsolateE + 288
7   ???                                 0x000000065bd042fd 0x0 + 27310179069
8   ???                                 0x000000065bdbd1d6 0x0 + 27310936534
```

**Comment by colingourlay** [comment]
The problem can be avoided by not disconnecting, which doesn't seem to have any negative impact at all. The segfault will remain a mystery (for now), but we have a workaround that's usable.

**Comment by colingourlay** [comment]
Fixed in #2
